sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -189,6 +189,7 @@ case class InsertIntoHadoopFsRelationCommand(
       sparkSession.catalog.refreshByPath(outputPath.toString)
 
       if (catalogTable.nonEmpty) {
+        sparkSession.sessionState.catalog.refreshTable(catalogTable.get.identifier)
         CommandUtils.updateTableStats(sparkSession, catalogTable.get)
```

Comment:
Can we switch line 193 and 192?

Reply:
There is a issue here: It may refresh the table twice. I moved this logical to `CommandUtils.updateTableStats`:
https://github.com/apache/spark/blob/2612167dad10eb6337344241db0c1614cde8e97a/sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala#L54-L55